magento/inventory
# Optimization to prevent oversell of products

New logic are introduced with reservation

### Description (*)
New reservation will be created before order placed (previous realization based on reservation after order placed). In case of errors within order placement additional reservation will be added to compensate


### Fixed Issues (if relevant)
https://jira.corp.magento.com/browse/MC-33831

### Manual testing scenarios (*)
When there is 1 quantity in stock and 2 orders placed to that product simultaneously both orders are going through and the saleable qty is updating to -1. Need to run simultaneously from 2 different browsers (or machines)

01. Add product to the cart which has 1 quantity left
02. Two customers placing orders at the same time for this SKU
03. Both orders will get success

EXPECTED RESULT
Only one order should go through

ACTUAL RESULT
Both orders going through and sellable quantity of the product updated to -1

Also please check functionality of the inventory:reservation:list-inconsistencies cli command with the next scenario: There are some reservations already exists, then update your msi with this changes, make some orders, check that command provide correct output. Place order with payment failure, check that command provide correct output

### Questions or comments

### Contribution checklist (*)
